### PR TITLE
[Issue #840] configurable products fix

### PR DESCRIPTION
**Pull request by ejnshtein**
Now configurable products can have customizable options.

To apply this fix you'll need this [PR with fix for quote-graphql](https://github.com/scandipwa/quote-graphql/pull/37).

Screenshot_20200825_184129
Screenshot_20200825_184209

### CODE REVIEW

File: src/app/component/CartItem/CartItem.component.js [hunk 1]
```diff
@@ -137,17 +137,18 @@ export class CartItem extends PureComponent {
...
         return (
-            <div
+            <span
               block="CartItem"
               elem="ItemOptionValue"
               key={ label }
             >
                 { label || value }
-            </div>
+{ array[i + 1] ? <>, </> : null }
```
Comment (alfredsgenkins) [inline src/app/component/CartItem/CartItem.component.js:150]: What the hell is this monstrosity? 

```js
array[i + 1] ? <>, </> : null
```

1. Why to use fragments around the `, `?
2. This is unreadable, prefer variable like `isNextAvailable`
3. I am against conditional rendering, but in this case the `isNextAvailable && ', '` would work better.

File: src/app/component/CartItem/CartItem.container.js [hunk 2]
```diff
@@ -148,13 +149,13 @@ export class CartItemContainer extends PureComponent {
             }
         } = this.props;
 
-        return variants.findIndex(({ sku }) => sku === itemSku);
+        return variants.findIndex(({ sku }) => sku === itemSku || itemSku.includes(sku));
```
Comment (alfredsgenkins) [inline src/app/component/CartItem/CartItem.container.js:152]: This check does not look right, child SKU might be literally any value.

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]

**Comment by alfredsgenkins** [comment]
Please also add https://github.com/scandipwa/quote-graphql/releases/tag/2.14.1 to composer dependenices.

File: src/app/component/CartItem/CartItem.container.js (hunk 2 above)
Comment (ejnshtein) [reply src/app/component/CartItem/CartItem.container.js:152]: itemSku in configurable products is dynamically generated when user adds the product to the cart.
Screenshot_20200826_143413_1
Screenshot_20200826_143834

Comment (ejnshtein) [reply src/app/component/CartItem/CartItem.container.js:152]: For example: I changed sku on test product blue and it changed in itemSku in user cart
Screenshot_20200826_150212

File: src/app/component/CartItem/CartItem.component.js (hunk 1 above)
Comment (ejnshtein) [reply src/app/component/CartItem/CartItem.component.js]: Yeah, jsx rule `react/jsx-one-expression-per-line` really wants to write this part like this.
I've made the isNextAvailabe variable, so should be better.

**Pull request merged**